src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -281,17 +282,38 @@ bool program_node::recalc_output_layout(bool invalidate_users_if_changed) {
 }
 
 bool program_node::is_dynamic() const {
+    // Strided slice loads data from {1,2,3} dependencies in impl::create method.
+    // It means that this data must be put into impl_params map
+    // Thus we treat it as "dynamic" case
+    // TODO: Remove once strided slice impl support runtime tensors for begin/end/stride
+    if (is_type<strided_slice>()) {
+        for (size_t i = 1; i < get_dependencies().size(); i++) {
+            if (!get_dependency(i).is_type<data>())
+                return true;
+        }
+    }
     for (const auto* input : get_dependencies()) {
-        if (input->get_output_layout().is_dynamic())
+        if (input->is_dynamic())
             return true;
     }
 
     return get_output_layout().is_dynamic();
 }
 
 bool program_node::is_dynamic() {
+    // Strided slice loads data from {1,2,3} dependencies in impl::create method.
+    // It means that this data must be put into impl_params map
+    // Thus we treat it as "dynamic" case
+    // TODO: Remove once strided slice impl support runtime tensors for begin/end/stride
+    if (is_type<strided_slice>()) {
+        for (size_t i = 1; i < get_dependencies().size(); i++) {
+            if (!get_dependency(i).is_type<data>())
+                return true;
+        }
+    }
+
     for (auto& input : get_dependencies()) {
-        if (input->get_output_layout(true).is_dynamic())
+        if (input->is_dynamic())
```

Comment:
isnt this recursive?

Reply:
You're right. And looks like it's not really needed anymore. Reverted.